x-pack/plugins/security/server/authentication/providers/pki.ts

```diff
@@ -307,6 +323,24 @@ export class PKIAuthenticationProvider extends BaseAuthenticationProvider {
         this.logger.debug('Self-signed certificate is detected in certificate chain');
         break;
       } else if (certificate.issuerCertificate === undefined) {
+        const protocol = request.socket.getProtocol();
+        if (!isRenegotiated && protocol && RENEGOTIATABLE_PROTOCOLS.includes(protocol)) {
+          this.logger.debug(
+            `Detected incomplete certificate chain with protocol '${protocol}', attempting to renegotiate connection.`
+          );
+          try {
+            await request.socket.renegotiate({ requestCert: true, rejectUnauthorized: false });
```

Comment:
Thanks for bringing it up. As discussed offline, I meant to point this out in the PR, but forgot to add a comment!

So the only way this distinction would make a difference is if the user has configured `server.ssl.clientAuthentication: required`, the socket is first authorized, and after renegotiation it is no longer authorized. It seems unlikely, but it could happen in theory if the client presents a different certificate upon renegotiation. Using a browser, renegotiation doesn't prompt me to select another certificate though. At any rate, if it did happen, the consequence is that we get some extra debug logs about the untrusted client certificate. 

Given that we don't currently have the capability to access this core config setting in a plugin, sounds like we are in agreement that the current approach is "good enough".